src/main.cpp

```diff
@@ -125,12 +125,46 @@ void runServer(Server *server)
 	delete socket;
 }
 
+// Candidate for removal after testing
+void printVector(const std::vector<std::string>* vecPtr)
+{
+	if (vecPtr)
+	{
+		for (std::vector<std::string>::const_iterator it = vecPtr->begin(); it != vecPtr->end(); ++it)
+		{
+			std::cout << *it;
+			if (it + 1 != vecPtr->end())
+				std::cout << ", ";
+		}
+		std::cout << std::endl;
+	}
+}
+
+void dansTestFunc()
+{
+	ConfigParser parser;
+	parser.parseConfig("config/default.conf");
+
+	// Access the servers
+	const std::vector<std::shared_ptr<Server>>& servers = parser.getServers();
+	for (std::vector<std::shared_ptr<Server>>::const_iterator it = servers.begin(); it != servers.end(); ++it) {
+		std::shared_ptr<Server> server = *it;
+		if (server)
+		{
+			std::cout << "Methods: ";
+			printVector(server->getLocationValue("/", "method"));
+			printVector(server->getLocationValue("/tmp", "method"));
+			printVector(server->getLocationValue("/cgi-bin", "directory"));
+		}
+	}
+}
+
 int main(int argc, char **argv)
 {
 	if (argc != 2 || !argv[1])
 	{
-		std::cerr << "The program needs one configuration file as input\n";
-		return 1;
+		dansTestFunc();
```

Comment:
I mean, sure, if you promise to put the original valid error message back once you're done with testing... :D

Reply:
Also the return as 1 should be changed back when you're done, as returning as 0 would indicate successful program run!